File: postgres_39/all_struct_with_default/Project.go
package dto

type Project struct {
	Project_id                string  `json:"project_id" default:"source"`
	Active_ind                *string `json:"active_ind" default:""`
	Complete_date             *string `json:"complete_date" default:""`
	Confidential_ind          *string `json:"confidential_ind" default:""`
	Confidential_release_date *string `json:"confidential_release_date" default:""`
	Description               *string `json:"description" default:""`
	Effective_date            *string `json:"effective_date" default:""`
	Expiry_date               *string `json:"expiry_date" default:""`
	Field_station_ind         *string `json:"field_station_ind" default:""`
	Land_right_ind            *string `json:"land_right_ind" default:""`
	Pden_ind                  *string `json:"pden_ind" default:""`
	Ppdm_guid                 *string `json:"ppdm_guid" default:""`
	Project_name              *string `json:"project_name" default:""`
	Project_num               *string `json:"project_num" default:""`
	Project_type              *string `json:"project_type" default:""`
	Remark                    *string `json:"remark" default:""`
	Seis_set_ind              *string `json:"seis_set_ind" default:""`
	Source                    *string `json:"source" default:""`
	Start_date                *string `json:"start_date" default:""`
	Strat_column_ind          *string `json:"strat_column_ind" default:""`
	Strat_interpretation_ind  *string `json:"strat_interpretation_ind" default:""`
	Well_ind                  *string `json:"well_ind" default:""`
	Row_changed_by            *string `json:"row_changed_by" default:""`
	Row_changed_date          *string `json:"row_changed_date" default:""`
	Row_created_by            *string `json:"row_created_by" default:""`
	Row_created_date          *string `json:"row_created_date" default:""`
	Row_effective_date        *string `json:"row_effective_date" default:""`
	Row_expiry_date           *string `json:"row_expiry_date" default:""`
	Row_quality               *string `json:"row_quality" default:""`
}

File: create_tables/project.py
import oracledb
import sys
import lib_platform as platform

if platform.system == "darwin":
    sys.path.insert(0, "database")
elif platform.system == "windows":
    sys.path.insert(0, "../database")

from conn import connection

# pw = getpass.getpass("Enter password: ")

# connection = oracledb.connect(
#     user="admin",
#     password="ppdm-3.9",
#     dsn="/kei-oracle-test.chkxr21l2l5s.ap-southeast-3.rds.amazonaws.com:1521/ORCL")

# print("Successfully connected to Oracle Database")

cursor = connection.cursor()


cursor.execute("""
    begin
        execute immediate 'drop table project';
        exception when others then if sqlcode <> -942 then raise; end if;
    end;""")

cursor.execute("""
    create table project (
        PROJECT_ID VARCHAR(40) PRIMARY KEY NOT NULL,
        ACTIVE_IND VARCHAR(1),
        COMPLETE_DATE TIMESTAMP(0),
        CONFIDENTIAL_IND VARCHAR(1),
        CONFIDENTIAL_RELEASE_DATE TIMESTAMP(0),
        DESCRIPTION VARCHAR(240),
        EFFECTIVE_DATE TIMESTAMP(0),
        EXPIRY_DATE TIMESTAMP(0),
        FIELD_STATION_IND VARCHAR(1),
        LAND_RIGHT_IND VARCHAR(1),
        PDEN_IND VARCHAR(1),
        PPDM_GUID VARCHAR(38),
        PROJECT_NAME VARCHAR(255),
        PROJECT_NUM VARCHAR(30),
        PROJECT_TYPE VARCHAR(40),
        REMARK VARCHAR(2000),
        SEIS_SET_IND VARCHAR(1),
        SOURCE VARCHAR(40),
        START_DATE TIMESTAMP(0),
        STRAT_COLUMN_IND VARCHAR(1),
        STRAT_INTERPRETATION_IND VARCHAR(1),
        WELL_IND VARCHAR(1),
        ROW_CHANGED_BY VARCHAR(30),
        ROW_CHANGED_DATE TIMESTAMP(0),
        ROW_CREATED_BY VARCHAR(30),
        ROW_CREATED_DATE TIMESTAMP(0),
        ROW_EFFECTIVE_DATE TIMESTAMP(0),
        ROW_EXPIRY_DATE TIMESTAMP(0),
        ROW_QUALITY VARCHAR(40))""")
